install/restart_scripts/renew_ca_cert.in

```diff
@@ -68,15 +70,38 @@ def _main():
             syslog.syslog(
                 syslog.LOG_NOTICE, "Stopped %s" % dogtag_service.service_name)
 
-    # Fetch the new certificate
-    db = certs.CertDB(api.env.realm, nssdir=paths.PKI_TOMCAT_ALIAS_DIR)
-    cert = db.get_cert_from_db(nickname)
-    if not cert:
-        syslog.syslog(syslog.LOG_ERR, 'No certificate %s found.' % nickname)
-        sys.exit(1)
-
     tmpdir = tempfile.mkdtemp(prefix="tmp-")
     try:
+        # If we are using HSM, create a pwd file with the password for
+        # the internal token + HSM
+        token_name = ca.get_token_name(nickname)
+        if token_name != INTERNAL_TOKEN:
+            if tasks.is_fips_enabled():
+                dbname = 'NSS FIPS 140-2 Certificate DB'
+            else:
+                dbname = 'NSS Certificate DB'
+            pwd_file = os.path.join(tmpdir, "pwd_file")
+            with open(pwd_file, "w") as pwd:
+                with open(paths.PKI_TOMCAT_PASSWORD_CONF, 'r') as fd:
+                    for line in fd:
+                        (token, pin) = line.split('=', 1)
+                        if token.startswith('hardware-'):
+                            token = token.replace('hardware-', '')
+                            pwd.write(f'{token}:{pin}')
+                        elif token == INTERNAL_TOKEN:
+                            pwd.write(f'{dbname}:{pin}')
+                pwd.flush()
```

Comment:
please add here `os.fchmod(pwd.fileno(), 0o600)` to protect this file from access.

Reply:
Thanks for the suggestion, done.